src/publication.rs

```diff
@@ -22,34 +22,42 @@ impl<'a> ToPayload for &'a [u8] {
     }
 }
 
+impl<'a> ToPayload for &'a str {
+    type Error = ();
+
+    fn serialize(&mut self, buffer: &mut [u8]) -> Result<usize, Self::Error> {
+        self.as_bytes().serialize(buffer)
+    }
+}
+
 impl<const N: usize> ToPayload for [u8; N] {
     type Error = ();
 
-    fn serialize(&self, buffer: &mut [u8]) -> Result<usize, ()> {
+    fn serialize(&mut self, buffer: &mut [u8]) -> Result<usize, ()> {
         (&self[..]).serialize(buffer)
     }
 }
 impl<const N: usize> ToPayload for &[u8; N] {
     type Error = ();
 
-    fn serialize(&self, buffer: &mut [u8]) -> Result<usize, ()> {
+    fn serialize(&mut self, buffer: &mut [u8]) -> Result<usize, ()> {
         (&self[..]).serialize(buffer)
     }
 }
 
-pub struct DeferredPublication<E, F: Fn(&mut [u8]) -> Result<usize, E>> {
+pub struct DeferredPublication<E, F: FnMut(&mut [u8]) -> Result<usize, E>> {
```

Comment:
Why not fnonce? We only ever call it once and we can consume the closure.

Reply:
This is needed by users like `minireq` (ref https://github.com/quartiq/minireq/pull/16)) -  The `FnMut` states that the _user_ can mutate objects within the closure. We have no knowledge of what the closure is, so the most relaxed trait bound would be `FnMut`

Reply:
Taking a `FnOnce` argument does not preclude it from mutate objects within its capture! That's not how these work. An `FnOnce` can mutate things just fine. Yes, `FnMut` is more relaced than `Fn`. But from your description and from how we use it, an `FnOnce` will do. Every `FnMut` is also `FnOnce`.

Reply:
Indeed, `FnOnce` does indeed look much nicer. I didn't realize the correlation between `Fn`, `FnMut`, `FnOnce` and `&self`, `&mut self`, and `self` until now, thanks for pointers in the right direction.